src/containerDeployment/containerDeploymentWebviewController.ts

```diff
@@ -166,6 +174,29 @@ export class ContainerDeploymentWebviewController extends FormWebviewController<
             return state;
         });
 
+        this.registerReducer("createConnectionGroup", async (state, payload) => {
+            const updatedState = (await createConnectionGroup(
+                payload.connectionGroupSpec,
+                this.mainController.connectionManager.connectionStore,
+                TelemetryViews.ContainerDeployment,
+                state,
+                state.formErrors,
+                state.formState,
+            )) as cd.ContainerDeploymentWebviewState;
+            this.updateState(updatedState);
+            return updatedState;
+        });
+
+        this.registerReducer("toggleConnectionGroupDialog", async (state) => {

```

Comment:
Slightly odd spot for a toggle; maybe add a paylod with `isOpen: boolean` prop.  Idempotence is nice to have when there's any risk of a command being issued multiple times.